documentation/restler/MakeTestProxyRecording.md

```diff
@@ -0,0 +1,167 @@
+}
+```
+
+Take the appConfiguration data-plane API as an example:
+- ServiceName="appconfiguration"
+- package="dataplane"
+
+So the contents of `assets.json` should look like:
+```json
+{
+  "AssetsRepo": "Azure/azure-sdk-assets",
+  "AssetsRepoPrefixPath": "apitest",
+  "TagPrefix": "apitest/appconfiguration/dataplane",
+}
+```
+
+Place this file in the `scenarios/` folder under the API version being tested.
+
+```
+specification/appconfiguration/data-plane
+├── Microsoft.AppConfiguration
+│   └── stable
+│       └── 1.0
+│           ├── appconfiguration.json
+│           ├── examples
+│           │   ├── CheckKeyValue.json
+│           │   ├── PutLock.json
+│           │   └── PutLock_IfMatch.json
+│           └── scenarios
+│               └── assets.json         <----- check-in assets.json here
+└── readme.md
+```
+
+## Start the test-proxy
+
+Open a new terminal window, cd to root of azure-rest-api-specs, and start test-proxy in this directory
+
+```bash
+test-proxy start
+```
+
+## Update the RESTler configuration file to use test-proxy
+
+The general instructions for running RESTler can be found [here](./QuickStart.md).
+There are some specific instructions for running RESTler with test-proxy.
+
+### Update engineSettings.json
+
+You need to modify the `engineSettings.json` file to use the test-proxy.
+
+```sh
+jq '.host = "localhost" | .target_port = 5000 | .no_ssl = true' restlerConfig/engine_settings.json > engine_settings.json
+```
+
+### Update the Restler custom dictionary with test-proxy headers
+
+The test-proxy requires custom headers to be sent with each request that tell it how to handle the request.
+One of these headers specifies the "recording id", which is obtained by sending a request to the test-proxy to start a new recording session.
+
+To start a recording session, send a `/Record/Start` request to the test-proxy.
+The request should specify the name of the recording file and the path to the `assets.json` file.
+The recording file will be created in the `.assets` folder in the root of the azure-rest-api-specs repo,
+but must have a unique name to avoid conflicts with other recordings.
+Here we make the name unique by prepending the same path as the `assets.json` file.
+The response of this request will contain an "x-recording-id" header. Save the value in `recording_id` variable.
+
+```sh
+scenarios=specifications/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/scenarios
+body='{ 
+  "x-recording-file": "'${scenarios}'/recording.json",
+  "x-recording-assets-file": "'${scenarios}'/assets.json"
+}'
+recording_id=$(curl -X POST -s -D - -d ${body} http://localhost:5000/Record/Start | grep 'x-recording-id' | awk '{print $2}' | sed 's/\r$//')
+```
+
+Now you can update the custom dictionary with the recording id and the test-proxy headers.
+
+```sh
+headers='{                                                 
+  "x-recording-upstream-base-uri": ["https://management.azure.com"],
```

Comment:
Would it be worth adding a sentence explaining that this base-uri is the original host for the service passed to RESTler (e.g. which was overwritten in the engine settings?)